atlp-rwanda/atlp-pulse-fn
# CRUD on grading system

# PR Description
fixing grading system page for admins or managers

# Description of tasks that were expected to be completed
- [x] view all grading system for organization
- [x] Make default grading system for organization
- [x] add new grading system for organization
- [x] delete grading system
- [x] view one grading system and it's details

# How has this been tested?
- For CLI
   - clone this [branch](https://github.com/atlp-rwanda/atlp-pulse-fn/tree/ft-grading-system) on with there is this changes 
   - run `npm test` for testing the unit test
- For Browser
   - clone this [branch](https://github.com/atlp-rwanda/atlp-pulse-fn/tree/ft-grading-system) or jump to the deployed version of this PL
   - login as admin or manager
      - admin: `emily.wallace@devpulse.com`
      - pass: `Test@12345`
   - in sidebar, click on `grading system`
   - you will view the list for all grading systems
   - **Make default grading system**
      - in the list of grading systems, all of the items has the button labeled `make default` except default one.
      - click on that button to make new default grading system
   - **add new:**
      - view on the top of the page, click on the button with label `Grading+` to add new grading system
      - the model will be displayed with the form
      - fill the form responsively 
   - **view one grading system**
      - First way
         - in the list of grading systems, for every item in list there is a button to labeled `view more` 
         - click that button to open that grading system
      - Second way
         - view on the top of the page, there is a select with label `select grading system`
         - click on it to show a drop down for all grading systems
         - select the one you want to view
   - **detete grading system**
      - once you open a grading system details
      - on the top right of the page, you'll find the button labeled `delete` 
      - once you click, it will open the model to confirm the delete action

# Number of Commits
**1 commit**

# Screenshots (If appropriate)
<img width="1116" alt="Screenshot 2023-08-04 at 09 09 03" src="https://github.com/atlp-rwanda/atlp-pulse-fn/assets/112763782/917b64bf-f390-4151-aa3c-bb4e064469d8">
<img width="617" alt="Screenshot 2023-08-03 at 13 08 17" src="https://github.com/atlp-rwanda/atlp-pulse-fn/assets/112763782/b7378bdf-d42e-4578-a266-cc2c2fc2a682">
<img width="592" alt="Screenshot 2023-08-03 at 13 08 54" src="https://github.com/atlp-rwanda/atlp-pulse-fn/assets/112763782/94a9de1b-df3a-4097-a2d0-dd9f4f72af68">
<img width="1113" alt="Screenshot 2023-08-04 at 09 16 58" src="https://github.com/atlp-rwanda/atlp-pulse-fn/assets/112763782/49ea1cfb-1d2c-4f37-b219-ae2e7eda241f">
<img width="491" alt="Screenshot 2023-08-04 at 09 18 00" src="https://github.com/atlp-rwanda/atlp-pulse-fn/assets/112763782/f21f75cd-2ce4-4bd6-bc73-d310dc4b776b">
<img width="487" alt="Screenshot 2023-08-04 at 09 18 25" src="https://github.com/atlp-rwanda/atlp-pulse-fn/assets/112763782/e6232ea6-844f-4096-bb1d-245a5468a8d8">

# Please check this Checklist before you submit your PR:
